tests/mu_convert_enum_test.c

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <string.h>
+#include "test_helper.h"
+
+#include "azure_macro_utils/macro_utils.h"
+
+#include "mu_convert_enum_test.h"
+
+// This lists the values in the first enum
+#define TEST_CONVERT_FROM_ENUM_VALUES \
+    test_from_a, \
+    test_from_b, \
+    test_from_c
+
+#define TEST_CONVERT_TO_ENUM_VALUES \
+    test_to_a, \
+    test_to_b
+
+MU_DEFINE_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_FROM_ENUM_VALUES);
+MU_DEFINE_ENUM(TEST_CONVERT_TO_ENUM, TEST_CONVERT_TO_ENUM_VALUES);
+
+/* The following code should not compile :
+MU_DEFINE_CONVERT_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM,
+    test_from_a, test_to_a,
+    test_from_c, test_to_b);*/
+
+MU_DEFINE_CONVERT_ENUM_WITH_VALIDATION(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM,
+    test_from_a, test_to_a,
+    test_from_b, test_to_b,
+    test_from_c, test_to_b);
+
+int run_mu_convert_enum_tests(void)
+{
+    int result = 0;
+
+    TEST_CONVERT_TO_ENUM result_value;
+    int convert_result;
+
+    convert_result = MU_CONVERT_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM)(test_from_a, &result_value);
+    POOR_MANS_ASSERT(convert_result == 0);
```

Comment:
>(convert_result == 0); [](start = 20, length = 22)

Missing test case where this fails (e.g. converting 42). Though I still wonder why we want to handle failures here and not just default to something (Invalid, or leave it as an option to define in the macro to use something like error)

Reply:
Added test.

---
In reply to: [394483229](https://github.com/Azure/azure-macro-utils-c/pull/36#discussion_r394483229) [](ancestors = 394483229)